src/react_native/share.cljs

```diff
@@ -3,8 +3,10 @@
     ["react-native-share" :default react-native-share]))
 
 (defn open
-  [options on-success on-error]
-  (-> ^js react-native-share
-      (.open (clj->js options))
-      (.then on-success)
-      (.catch on-error)))
+  ([options]
+   (open options nil nil))
+  ([options on-success on-error]
+   (-> ^js react-native-share
+       (.open (clj->js options))
+       (.then on-success)
```

Comment:
this seems slightly "dangerous" - on-success and on-error can be `nil`. I think you should default them to empty functions instead? i.e `(open options #() #()))` or put some check to make sure they're functions before being called

Reply:
I agree, why do we want to open it without an on-success or on-error call?

as @J-Son89  said, at least call it with empty functions to avoid exceptions :+1: